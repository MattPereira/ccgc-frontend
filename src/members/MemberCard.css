.MemberCard-link {
    text-decoration: none;
    color: black;
}


.MemberCard-card:hover { /* <Card /> that is also a <Link /> */
    text-decoration: none;
    color: #198754;
    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
  }

.MemberCard-title {
    background-color: #0a8ce3;
}