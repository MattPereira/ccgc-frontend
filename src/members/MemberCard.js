import React from "react";
import { Link } from "react-router-dom";
import { Card, CardBody, CardTitle, CardText } from "reactstrap";
import "./MemberCard.css";

/** Member card component.
 *
 * Show information about a member and function as link to member's detail page.
 *
 * MemberCard is rendered by MemberList to show a "card" for each member.
 *
 * MemberList -> MemberCard
 */

const MemberCard = ({ username, email, firstName, lastName, bio }) => {
  return (
    <div className="mb-3 text-center">
      <Link to={`/members/${username}`} className="MemberCard-link">
        <Card body className="MemberCard-card">
          <CardBody>
            <CardTitle tag="h5">
              {firstName} {lastName}
            </CardTitle>
            <CardText className="mb-2">{email}</CardText>
            <CardText>
              <i>{bio}</i>
            </CardText>
          </CardBody>
        </Card>
      </Link>
    </div>
  );
};

export default MemberCard;
