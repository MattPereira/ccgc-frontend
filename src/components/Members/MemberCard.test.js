import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import MemberCard from "./MemberCard";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <MemberCard
        username="Happy Gilmore"
        firstName="Happy"
        lastName="Gilmore"
        avgStrokes="77"
        avgPutts="22"
        avgGreenies="1.11"
      />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <MemberCard
        username="Happy Gilmore"
        firstName="Happy"
        lastName="Gilmore"
        avgStrokes="77"
        avgPutts="22"
        avgGreenies="1.11"
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("displays member card with average strokes, putts, and greenies", function () {
  render(
    <MemoryRouter>
      <MemberCard
        username="Happy Gilmore"
        firstName="Happy"
        lastName="Gilmore"
        avgStrokes="77"
        avgPutts="22"
        avgGreenies="1.11"
      />
    </MemoryRouter>
  );

  const cardTitle = screen.getByText("Happy Gilmore");
  expect(cardTitle).toBeInTheDocument();

  const cardTableHead = screen.getByText("Average");
  expect(cardTableHead).toBeInTheDocument();

  const strokesAvg = screen.getByText("77");
  expect(strokesAvg).toBeInTheDocument();

  const puttsAvg = screen.getByText("22");
  expect(puttsAvg).toBeInTheDocument();

  const greeniesAvg = screen.getByText("1.11");
  expect(greeniesAvg).toBeInTheDocument();
});
