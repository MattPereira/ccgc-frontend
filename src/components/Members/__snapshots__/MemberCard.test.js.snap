// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-5 text-center"
  >
    <a
      class="MemberCard-link"
      href="/members/Happy Gilmore"
    >
      <div
        class="MemberCard-card card"
      >
        <div
          class="text-white py-3 mb-0 bg-dark card-title h5"
          tag="h5"
        >
          Happy Gilmore
        </div>
        <div
          class="card-body"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-10"
            >
              <table
                class="table"
              >
                <thead>
                  <tr>
                    <th
                      colspan="2"
                    >
                      Average
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>
                      Strokes
                    </td>
                    <td>
                      77
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Putts
                    </td>
                    <td>
                      22
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Greenies
                    </td>
                    <td>
                      1.11
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
